[Cocoa] Remove permissions for SYS_setsockopt from the WebContent process

brentfulgham opened the pull request:
#### c9647ae53c5d528fc86cb2a37905683ccb7b0422
<pre>
[Cocoa] Remove permissions for SYS_setsockopt from the WebContent process
<a href="https://bugs.webkit.org/show_bug.cgi?id=265413">https://bugs.webkit.org/show_bug.cgi?id=265413</a>
&lt;<a href="https://rdar.apple.com/107933292">rdar://107933292</a>&gt;

Reviewed by Per Arne Vollan.

We needed to keep access to `setsockopt` in the WebContent process until Content
Filtering had completely moved to the Network process. That happened in macOS 13,
so we should be able to remove this ability for Ventura and newer operating systems.
All iOS ports have made this transition already, so no version checks are needed there.

* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
* Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:

Canonical link: <a href="https://commits.webkit.org/271239@main">https://commits.webkit.org/271239@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c9644e188d6976a83dc42a246d3ca42286ab1151

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/27747 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/6385 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/28992 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/29963 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/25351 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/8337 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/3778 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/25117 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/28012 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/5141 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/23809 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/4464 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/4629 "Passed tests") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/30603 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/25344 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/25249 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/30787 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/4648 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/2794 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/28709 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/6151 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/5095 "Built successfully") | | | 
| [  ~~🛠 🧪 unsafe-merge~~](https://ews-build.webkit.org/#/builders/22/builds/3594 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/5086 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pvollan commented on Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:
```diff
@@ -1876,8 +1876,10 @@
         SYS_munlock
         SYS_openat_nocancel
         SYS_proc_rlimit_control
+#if __MAC_OS_X_VERSION_MIN_REQUIRED < 130000
```
I think we may need to add PLATFORM(MAC) to this condition.

brentfulgham replied:
Ah, good point! I will fix that shortly.

pvollan replied:
One option is to leave the syscall in for Catalyst:

#if !PLATFORM(MAC) || __MAC_OS_X_VERSION_MIN_REQUIRED < 130000

brentfulgham replied:
I don't think it's needed for Catalyst, since it is already blocked for iOS and we have no telemetry for it.

pvollan replied:
Ah, good point!

pvollan reviewed (approved):
Very nice! R=me.

brentfulgham committed:
[Cocoa] Remove permissions for SYS_setsockopt from the WebContent process
https://bugs.webkit.org/show_bug.cgi?id=265413
<rdar://107933292>

Reviewed by Per Arne Vollan.

We needed to keep access to `setsockopt` in the WebContent process until Content
Filtering had completely moved to the Network process. That happened in macOS 13,
so we should be able to remove this ability for Ventura and newer operating systems.
All iOS ports have made this transition already, so no version checks are needed there.

* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
* Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:

Canonical link: https://commits.webkit.org/271239@main
